### PR TITLE
Add option for image family URI as source and fix smoke test issue for no external IP option

### PR DESCRIPTION
**Pull request by lionelds-wrk**
New parameter added to use an image family URI as the source. Also, fixed an issue with the smoke test, which would break when the --no-external-ip option was used.

### CODE REVIEW

**Comment by lionelds-wrk** [comment]
@googlebot I signed it!

File: custom_image_utils/shell_script_generator.py [hunk 1]
```diff
@@ -62,12 +62,19 @@
   done
 
   echo 'Creating disk.'
+  if [ '{source_image_family_uri}' = '' -o  '{source_image_family_uri}' = 'None' ]; then
```
Comment (functicons) [inline custom_image_utils/shell_script_generator.py:65]: nit: we prefer this syntax `[[ '{source_image_family_uri}' == '' ||  '{source_image_family_uri}' == 'None' ]]`.

File: tests/test_shell_script_generator.py [hunk 2]
```diff
@@ -114,7 +121,8 @@ def test_generate_shell_script(self):
         'oauth': '',
         'project_id': 'my-project',
         'storage_location': 'us-east1',
-        'shutdown_timer_in_sec': 500
+        'shutdown_timer_in_sec': 500,
+        'source_image_family_uri': 'projects/my-dataproc-project/global/images/family/debian-10'
```
Comment (functicons) [inline tests/test_shell_script_generator.py:125]: Do you expect the user to use an image family from their own project?

**Comment by lionelds-wrk** [comment]
> Thanks for the contribution! But I'm not so clear about the use case of this new flag. Is it supposed to point to a family of released Dataproc images or a family of the user's Dataproc custom images?

It will point to a family of user's Dataproc images. The use case is that one team will apply additional customization's to the family of Dataproc images created by another team..

**Comment by functicons** [comment]
That makes sense.

File: README.md [hunk 3]
```diff
@@ -47,11 +47,15 @@ python generate_custom_image.py \
...
     understand Dataproc versioning, please refer to
     [documentation](https://cloud.google.com/dataproc/docs/concepts/versioning/overview).
-    **This argument is mutually exclusive with --base-image-uri**.
+    **This argument is mutually exclusive with --base-image-uri and --source-image-family**.
 *   **--base-image-uri**: The full image URI for the base Dataproc image. The
     customization script will be executed on top of this image instead of an
     out-of-the-box Dataproc image. This image must be a valid Dataproc image.
-    **This argument is mutually exclusive with --dataproc-version.**
+    **This argument is mutually exclusive with --dataproc-version and --source-image-family.**
+*   **--source-image-family**: The image family that the boot disk will be
```
[the remark this answers is not in the document]
Comment (lionelds-wrk) [reply README.md]: Changed the option to --base-image-family, added command to get list of images/families and an example value.

File: custom_image_utils/args_inferer.py [hunk 4]
```diff
@@ -78,11 +88,41 @@ def _get_dataproc_image_version(image_uri):
...
+    if pipe.returncode != 0:
+      raise RuntimeError(
+          "Cannot find dataproc base family image, please check and verify "
+          "the family URI.")
+
+    temp_file.seek(0)  # go to start of the stdout
+    stdout = temp_file.read()
+    # parse the first ready image with the dataproc version attached in labels
+    if stdout:
+      parsed_line = stdout.decode('utf-8').strip()  # should be just one value
```
[the remark this answers is not in the document]
Comment (lionelds-wrk) [reply custom_image_utils/args_inferer.py]: Changed the variable name.

File: custom_image_utils/args_inferer.py [hunk 5]
```diff
@@ -78,11 +88,41 @@ def _get_dataproc_image_version(image_uri):
   raise RuntimeError("Cannot find dataproc base image: %s", image_uri)
 
 
+def _get_dataproc_image_family_version(image_family_uri):
```
[the remark this answers is not in the document]
Comment (lionelds-wrk) [reply custom_image_utils/args_inferer.py]: Changed the function name.

File: custom_image_utils/shell_script_generator.py (hunk 1 above)
Comment (lionelds-wrk) [reply custom_image_utils/shell_script_generator.py]: Updated syntax.

File: tests/test_shell_script_generator.py (hunk 2 above)
Comment (lionelds-wrk) [reply tests/test_shell_script_generator.py]: Yes. See use case in previous conversation.

**Pull request merged**